cmake/FindLibBfd.cmake

```diff
@@ -48,12 +44,40 @@ mark_as_advanced(LIBBFD_INCLUDE_DIRS LIBBFD_LIBRARIES)
 if(${LIBBFD_FOUND})
 find_package(LibOpcodes)
 SET(CMAKE_REQUIRED_LIBRARIES ${LIBBFD_LIBRARIES} ${LIBOPCODES_LIBRARIES})
-# libbfd.a is not statically linked with libiberty.a or libz.a so we must manually
-# do it. Furthermore, libbfd uses some libc symbols that we must manually
-# link against if we're not using static libc (which includes such symbols).
-if(STATIC_LINKING)
+if(STATIC_LINKING OR LIBBFD_STATIC)
+  # libbfd.so is linked with the required libraries but libbfd.a is not.
+  # So if we do a static bpftrace build or libbfd.so is not available (e.g. on
+  # openSUSE), we must link them manually.
+  # Furthermore, libbfd uses some libc symbols that we must manually link
+  # against if we're not using static libc (which includes such symbols).
   find_package(ZLIB)
-  list(APPEND CMAKE_REQUIRED_LIBRARIES ${LIBIBERTY_LIBRARIES} ${LIBZ_LIBRARIES})
+  find_library (LIBIBERTY_LIBRARIES
+    NAMES
+      iberty
+    PATHS
+      ENV LIBRARY_PATH
+      ENV LD_LIBRARY_PATH)
+  find_library (LIBSFRAME_LIBRARIES
+    NAMES
+      sframe
+    PATHS
+      ENV LIBRARY_PATH
+      ENV LD_LIBRARY_PATH)
+  find_library (LIBZSTD_LIBRARIES
+    NAMES
+      zstd
+    PATHS
+      ENV LIBRARY_PATH
+      ENV LD_LIBRARY_PATH)
```

Comment:
Perhaps output a STATUS message that additional libraries are required if only static libbfd is available? I don't think we document these dependencies

Reply:
Good idea. I added the message, although I didn't specify the exact list b/c some distros do not require all of them. At least the user now knows what's going on and can look into particular `CMakeLists.txt` in case of a problem.